### PR TITLE
feat(iconv): implement iconv and encodings : ASCII, (unicode defined encodings)utf8, utf16, utf32

### PR DESCRIPTION
**Pull request by rishadbaniya**
Instead of following a dynamic library loading method(which GNU does) we have gone for static library(since we have small set of encodings right now), this adds on less complexity for dev purpose as well,

**FYI**: we can always move to a dynamic library approach in the future if we need...as the encoding follows the following flow

```FROM ENCODING -> UCS-4 FORMAT(Intermediate format in our case) -> TO ENCODING ```

### CODE REVIEW

File: i18n/Cargo.toml [hunk 1]
```diff
@@ -12,10 +12,16 @@ clap.workspace = true
 gettext-rs.workspace = true
 bytemuck = { version = "1.17.0", features = ["derive"] }
 byteorder = "1.5.0"
+strum = "0.26.3"
+strum_macros = "0.26.4"
```
Comment (jgarzik) [inline i18n/Cargo.toml]: Delete the micro version of all deps.  e.g.
"0.26.3" -> "0.26"
"0.17.0" -> "0.17" 
etc.

File: i18n/iconv.rs [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+        Some(files) => files
+            .into_iter()
+            .map(|file| plib::io::input_stream(&file, true))
+            .collect::<Result<Vec<_>, _>>()?,
+        None => vec![Box::new(io::stdin().lock())],
+    };
+
+    for mut input in inputs {
+        let mut inp_buf = Vec::new();
+        input.read_to_end(&mut inp_buf)?;
```
Comment (jgarzik) [inline i18n/iconv.rs]: What level of difficulty is a streaming implementation?

An all-in-memory implementation limits users to documents to a maximum of 1/2 available memory + swap.

**Review by jgarzik — COMMENTED** [review]
Main comment is about a non-streaming implementation.  Is streaming easy or difficult?  

For the user, reading everything into memory is a challenge for very large files.

File: i18n/iconv.rs (hunk 2 above)
Comment (rishadbaniya) [reply i18n/iconv.rs]: well, a hurdle for me was getting the boundary of the last character of a specific encoding(especially utf8 coz of it's variable length)
i'll take a look into it's solution today

Comment (rishadbaniya) [reply i18n/iconv.rs]: actually, i can probably work "directly" with the Read trait object and extracting just what i need throughout the conversion

Comment (rishadbaniya) [reply i18n/iconv.rs]: hmm..doesn't seem simple ..i'll try to figure out a way

**Comment by rishadbaniya** [comment]
i've replaced with chain of iterator consumption..hence large files will work without heavy memory consumption

**Review by jgarzik — APPROVED** [review]

**Pull request merged**